core/Pos/Core/Genesis/Types.hs

```diff
@@ -113,13 +111,13 @@ bootDustThreshold (GenesisWStakeholders bootHolders) =
     -- it's safe to use it here because weights are word16 and should
     -- be really low in production, so this sum is not going to be
     -- even more than 10-15 coins.
-    unsafeIntegerToCoin . sum $ map fromIntegral $ HM.elems bootHolders
+    unsafeIntegerToCoin . sum $ map fromIntegral $ toList bootHolders
 
 -- | Safe constructor for 'GenesisCoreData'. Throws error if something
 -- goes wrong.
 mkGenesisCoreData ::
        [AddrDistribution]
-    -> HashMap StakeholderId Word16
+    -> Map StakeholderId Word16
     -> Either String GenesisCoreData
 mkGenesisCoreData distribution bootStakeholders = do
```

Comment:
Ok, I will add this check just in case, but at the same time I admit there may be cases when someone may want to have different distribution in genesis just to test something.
Anyway, if someone needs it, they will comment this check out or come up with something better.